tempesta_fw/http_sess.c

```diff
@@ -402,51 +438,246 @@ tfw_http_sticky_add(TfwHttpResp *resp)
 	return r;
 }
 
+/*
+ * Calculate HMAC value for redirection mark.
+ *
+ * HMAC value is based on:
+ * - Number of attempts for session establishing;
+ * - Timestamp of first attempt;
+ * - Secret key.
+ */
+static int
+__redir_hmac_calc(TfwHttpReq *req, RedirMarkVal *mv)
+{
+	int r;
+	char desc[sizeof(struct shash_desc)
+		  + crypto_shash_descsize(tfw_sticky_shash)]
+		  CRYPTO_MINALIGN_ATTR;
+	struct shash_desc *shash_desc = (struct shash_desc *)desc;
```

Comment:
There is standard macro `SHASH_DESC_ON_STACK` for the 2 declarations. Please update `__sticky_calc()` as well.

Reply:
Done.